private/highscoreList.js

```diff
@@ -0,0 +1,95 @@
+const router = require('express').Router();
+const db = require('../database/connection.js').connection;
+
+router.get('/api/highscores/user/:userHighestScore', (req, res) => {
+  db.query(`(SELECT high_score.high_score_id, player.username, high_score.score, high_score.date_time 
+        FROM player JOIN high_score 
+        ON player.player_id = high_score.player_id 
+        WHERE score >= ?
+        ORDER BY score ASC
+        LIMIT 5)
+    
+    UNION ALL
+    
+    (SELECT high_score.high_score_id, player.username, high_score.score, high_score.date_time 
+        FROM player JOIN high_score 
+        ON player.player_id = high_score.player_id
+        WHERE score < ?
+        ORDER BY score DESC
+        LIMIT 5);`, [req.params.userHighestScore, req.params.userHighestScore], (error, result, fields) => {
+    if (result && result.length) {
+      const highscores = [];
+      for (const highscore of result) {
+        highscores.push({
+          place: highscore.high_score_id, username: highscore.username, score: highscore.score, dateTime: highscore.date_time,
+        });
+      }
+      res.send({
+        highscores,
+      });
+    } else {
+      res.send({
+        message: 'No scores found',
+      });
+    }
+  });
+});
+
+router.get('/api/highestscore/user/:userId', (req, res) => {
+  db.query('SELECT score FROM high_score WHERE player_id = ? ORDER BY score DESC LIMIT 1;', [req.params.userId], (error, result, fields) => {
+    if (result && result.length) {
+      res.send({
+        highestscore: result[0].score,
+      });
+    } else {
+      res.send({
+        message: 'No scores found.',
+      });
+    }
+  });
+});
+
+router.get('/api/highestscores/:currentPage', (req, res) => {
+  db.query('SELECT * FROM high_score;',(error, result, fields) => {
+    if (result && result.length) {
+      const pageLimit = 1;
+      const currentPage = req.params.currentPage;
+      const scoresSize = result.length;
+      const offset = (currentPage - 1) *pageLimit; //minus one because offset defines from which row we want to retrieve data
+      const fullPages = scoresSize/pageLimit;
+      const reminderOfScores = scoresSize % pageLimit === 0 ? 0 : 1;
+      const pages = fullPages + reminderOfScores;
+      db.query(`SELECT high_score.high_score_id, player.username, high_score.score, high_score.date_time 
```

Comment:
You might need order by high_score.score here. To make it more efficient we don't need to store scores in the database in correct order as it would need to overwrite a lot of entries if one of the upper places changed. Similarly as in the article that I put to jira before. But then when you query then you need to order them else they will just be random.

Reply:
So also then the place won't be the id but we will need to calculate it. But based on page and pagelength you can easily get the number of that.
Because imagine that we have 100 entries and now there is someone who gets the top 1 score. We would need to remove the 100th entry and then for the rest of the 99 ones change the id one by one which is a lot? In case if we decided to store more than 100 it would only get worse.

Reply:
yep